timer.go

```diff
@@ -51,3 +54,6 @@ func (f *fakeTimer) expire(now time.Time) *time.Duration {
 	}
 	return nil
 }
+
+// Sleep momentarily so that other goroutines can process.
```

Comment:
Please delete this. We can't introduce arbitrary sleeps in test code.

Reply:
Hmm, that is a little bit of a code smell. Working with clocks and goroutines make it easy to introduce race conditions, and I want to avoid that. Let me take a look at the problem we are trying to solve again.

PS - Sorry this review took so long. :(

Reply:
I wanted to share my initial thoughts in https://github.com/jonboulle/clockwork/pull/92. Obviously the `With*` functions are not finalized and I just realized `afterTime` isn't necessary. Regardless, I think my major takeaways are:
1. We need to provide a way for the user that created the context to cancel the context they receive, even if we aren't respecting parent cancellation.
2. I *think* we can do this with only a goroutine and 2 select statements(?)

I haven't started testing yet, so maybe I haven't hit the same issue as you that seems to require the sleep?

LMK if anything stands out to you. Going to continue working through it over the weekend.

Reply:
Please take a look at https://github.com/jonboulle/clockwork/pull/92. As far as I am aware, it has no race conditions and although we do have to sleep at 1 point in 2 different test cases, those cases are run in parallel and that sleep is not in the library's code path.

I'll let that PR hang for comments for ~2 weeks. LMK if that gives you everything you need or if you have any concerns.

Reply:
I added a comment on usage that is addressed by my PR but is not addressed in #92.

Let me know what you think.

Meanwhile, I'll try to work around the time.Sleep issue.